x-pack/plugin/mapper-counted-keyword/src/test/java/org/elasticsearch/xpack/countedkeyword/CountedKeywordFieldMapperTests.java

```diff
@@ -64,9 +71,103 @@ protected Object generateRandomInputValue(MappedFieldType ft) {
+    }
+
+    public void testSyntheticSourceManyNullValue() throws IOException {
+        DocumentMapper mapper = createSytheticSourceMapperService(mapping(b -> {
+            b.startObject("field");
+            minimalMapping(b);
+            b.field("synthetic_source_keep", "none");
+            b.endObject();
+        })).documentMapper();
+
+        int nullCount = randomIntBetween(1, 5);
+
+        String expected = "{}";
+        CheckedConsumer<XContentBuilder, IOException> buildInput = b -> {
+            b.startArray("field");
+            for (int i = 0; i < nullCount; i++) {
+                b.nullValue();
+            }
+            b.endArray();
+        };
+
+        assertThat(syntheticSource(mapper, buildInput), equalTo(expected));
+        assertThat(syntheticSource(mapper, new SourceFilter(new String[] { "field" }, null), buildInput), equalTo(expected));
+        assertThat(syntheticSource(mapper, new SourceFilter(null, new String[] { "field" }), buildInput), equalTo("{}"));
+    }
+
+    @Override
+    public void testSyntheticSourceKeepAll() throws IOException {
+        // For now, native synthetic source is only supported when "synthetic_source_keep" mapping attribute is "none"
+    }
+
+    @Override
+    public void testSyntheticSourceKeepArrays() throws IOException {
+        // For now, native synthetic source is only supported when "synthetic_source_keep" mapping attribute is "none"
+    }
+
+    @Override
+    public void testSyntheticSourceKeepNone() throws IOException {
+        // For now, native synthetic source is only supported when "synthetic_source_keep" mapping attribute is "none"
+    }
+
     @Override
     protected SyntheticSourceSupport syntheticSourceSupport(boolean ignoreMalformed) {
-        throw new AssumptionViolatedException("not supported");
+        return new SyntheticSourceSupport() {
+            @Override
+            public SyntheticSourceExample example(int maxValues) throws IOException {
+                if (randomBoolean()) {
+                    Tuple<String, String> v = generateValue();
+                    return new SyntheticSourceExample(v.v1(), v.v2(), this::mapping);
+                }
+                int maxNullValues = 5;
+                List<Tuple<String, String>> values = randomList(1, maxValues, this::generateValue);
+                List<String> in = Stream.concat(values.stream().map(Tuple::v1), randomList(0, maxNullValues, () -> (String) null).stream())
+                    .toList();
+
+                in = shuffledList(in);
+
+                List<String> outList = values.stream().map(Tuple::v2).sorted().toList();
+
+                Object out = outList.size() == 1 ? outList.get(0) : outList;
+                return new SyntheticSourceExample(in, out, this::mapping);
+            }
+
+            private Tuple<String, String> generateValue() {
+                String v = ESTestCase.randomAlphaOfLength(5);
```

Comment:
Should we explicitly add some duplicate values here since this is one of the main features of the field? This code would do that with some probability but i think it would be nice to be explicit about it.

Reply:
Good point, I'll include the updates in the follow-up PR.